docs/process/release.md

```diff
@@ -16,15 +16,20 @@ Once the release is complete, the assigned release driver will:
   * Documentation
   * Coverage
   * Performance / Memory / Size benchmarks
+  * Cargo.toml files
+    * Cargo.toml files need to specify versions for each dependency.
```

Comment:
Suggestion: we could exclude dev-dependencies here

Reply:
From what @sffc has said, we should explicitly not use versions for `dev-dependencies` because that will avoid the problem with having to edit Cargo.toml files to break dependency cycles, so I'll adjust the text here to reflect that.